Silence warnings from 3rd-party CocoaPods dependencies

wtimme opened the pull request:
### Introduction

This pull request configures CocoaPods in the way that warnings from 3rd-party code are not displayed in Xcode. This has two huge benefits:

- Seeing the number of warnings going down is a very, _very_ nice feeling
- We only see warnings for things that we can fix, right there, in our code base

Here's a clean build on my machine:

| `develop` right now | This branch |
| ------ | ----- |
| <img width="273" alt="element-warnings-before" src="https://user-images.githubusercontent.com/1681085/169898099-7ccd7f75-5b44-4799-8cba-8dba698136f8.png"> | <img width="273" alt="element-warnings-after" src="https://user-images.githubusercontent.com/1681085/169898237-784bb33d-94cc-4957-8962-e712d15e2113.png"> |

The one instance that one might want to see warnings is when using a "local" version of the matrix-ios-sdk. I was unsure whether this is something that you want/need, so I did not add it. Should not be much effort though, so let me know what you think.

Signed-off-by: Wolfgang Timme <element-signoff@milchbartstrasse.de>

### Pull Request Checklist

- [x] I read the [contributing guide](https://github.com/vector-im/element-ios/blob/develop/CONTRIBUTING.md)
- [x] UI change has been tested on both light and dark themes, in portrait and landscape orientations and on iPhone and iPad simulators
- [x] Accessibility has been taken into account.
* [x] Pull request is based on the develop branch
- [x] Pull request contains a [changelog file](https://github.com/matrix-org/matrix-ios-sdk/blob/develop/CONTRIBUTING.md#changelog) in ./changelog.d
- [x] You've made a self review of your PR
- [x] Pull request includes screenshots or videos of UI changes
- [x] Pull request includes a [sign off](https://github.com/matrix-org/matrix-ios-sdk/blob/develop/CONTRIBUTING.md#sign-off)

pixlwave commented:
Hey @wtimme thanks for taking the time to open this PR, it sound great. As you mentioned it would indeed be nice for us to still see warnings for MatrixSDK as we are responsible for that. Additionally it would be nice to see them for AnalyticsEvent as that's one of ours and I was thinking about Olm too but I see they're already disabled on that due to it being "bad" so probably a good idea to leave that alone 😅

Additionally would you be able to a) add a [sign off](https://github.com/matrix-org/matrix-ios-sdk/blob/develop/CONTRIBUTING.md#sign-off) to your PR and b) update the changelog filename to `pr-6196.build` as it doesn't have an issue.

pixlwave commented:
Hey @wtimme thanks for taking the time to open this PR, it sound great. As you mentioned it would indeed be nice for us to still see warnings for MatrixSDK as we are responsible for that. Additionally it would be nice to see them for AnalyticsEvent as that's one of ours and I was thinking about Olm too but I see they're already disabled on that due to it being "bad" so probably a good idea to leave that alone 😅

Additionally would you be able to a) add a [sign off](https://github.com/matrix-org/matrix-ios-sdk/blob/develop/CONTRIBUTING.md#sign-off) to your PR and b) update the changelog filename to `pr-6196.build` as it doesn't have an issue.

wtimme commented:
Thanks for your feedback, @pixlwave! I have updated the pull request's description to include a Sign-Off as per your comment.

I have also added `:inhibit_warnings => false` for the dependencies `MatrixSDK` and `AnalyticsEvents`.

Here's another comparison after updating the `Podfile`:

| `develop` (before) | this pull request's branch |
| ------------------ | -------------------------- |
| <img width="330" alt="element-warnings-develop" src="https://user-images.githubusercontent.com/1681085/171140885-53919ebe-78b9-4a8f-bdbc-40a93b73ea63.png"> | <img width="328" alt="element-warnings-after" src="https://user-images.githubusercontent.com/1681085/171140955-767d7ab0-5785-4e29-bcb1-6dec86c1557c.png"> |

pixlwave commented:
Thanks for your contribution Wolfgang 🙏
